src/Aspire.Hosting.Azure.EventHubs/AzureEventHubsExtensions.cs

```diff
@@ -99,4 +101,105 @@ public static IResourceBuilder<AzureEventHubsResource> AddEventHub(this IResourc
         builder.Resource.Hubs.Add((name, configureHub));
         return builder;
     }
+
+    /// <summary>
+    /// Configures an Azure Event Hubs resource to be emulated. This resource requires an <see cref="AzureEventHubsResource"/> to be added to the application model.
+    /// </summary>
+    /// <param name="builder">The Azure Event Hubs resource builder.</param>
+    /// <param name="storageResource">An optional Azure Storage resource.</param>
+    /// <param name="configureContainer">Callback that exposes underlying container used for emulation to allow for customization.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<AzureEventHubsResource> RunAsEmulator(this IResourceBuilder<AzureEventHubsResource> builder, IResourceBuilder<AzureStorageResource>? storageResource = null, Action<IResourceBuilder<AzureEventHubsEmulatorResource>>? configureContainer = null)
```

Comment:
How do you feel about this @mitchdenny? Allowing the storage resource to be passed in? Should we just always add one instead?

Reply:
Since we are trying to run the emulator locally I don't feel like it matters to pass a well-known instance, almost like an implementation detail and creating separate container should not be much of a burden, not much more resources are taken I believe. Also this can always be added later if users ask for it.

Reply:
Lets remove it.